iot/manager/manager.go

```diff
@@ -690,6 +652,166 @@ func sendCommand(w io.Writer, projectID string, region string, registry string,
 
 // [END iot_send_command]
 
+// START BETA FEATURES
+
+// [START iot_create_gateway]
+
+// createGateway creates a new IoT Core gateway with a given id, public key, and auth method.
+// gatewayAuthMethod can be one of: ASSOCIATION_ONLY, DEVICE_CREDENTIALS_ONLY, ASSOCIATION_AND_DEVICE_AUTH_TOKEN
+func createGateway(w io.Writer, projectID string, region string, registryID string, gatewayID string, gatewayAuthMethod string, publicKeyPath string) (*cloudiot.Device, error) {
+	client, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	keyBytes, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	gateway := cloudiot.Device{
+		Id: gatewayID,
+		Credentials: []*cloudiot.DeviceCredential{
+			{
+				PublicKey: &cloudiot.PublicKeyCredential{
+					Format: "RSA_X509_PEM",
+					Key:    string(keyBytes),
+				},
+			},
+		},
+		GatewayConfig: &cloudiot.GatewayConfig{
+			GatewayType:       "GATEWAY",
+			GatewayAuthMethod: gatewayAuthMethod,
+		},
+	}
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	response, err := client.Projects.Locations.Registries.Devices.Create(parent, &gateway).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Fprintln(w, "Successfully created gateway:", gatewayID)
+
+	return response, err
+}
+
+// [END iot_create_gateway]
+
+// [START iot_list_gateways]
+
+// listGateways lists all the gateways in a specific registry
+func listGateways(w io.Writer, projectID string, region string, registryID string) ([]*cloudiot.Device, error) {
+	client, err := getClient()
+	if err != nil {
+		return nil, err
+	}
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	response, err := client.Projects.Locations.Registries.Devices.List(parent).GatewayListOptionsGatewayType("GATEWAY").Do()
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(response.Devices) == 0 {
+		fmt.Fprintln(w, "No gateways found")
+	} else {
+		fmt.Fprintln(w, len(response.Devices), "devices:")
+		for _, gateway := range response.Devices {
+			fmt.Fprintf(w, "\t%s\n", gateway.Id)
+		}
+	}
+
+	return response.Devices, err
+}
+
+// [END iot_list_gateways]
+
+// [START iot_bind_device_to_gateway]
+
+// bindDeviceToGateway creates an association between an existing device and gateway
+func bindDeviceToGateway(w io.Writer, projectID string, region string, registryID string, gatewayID string, deviceID string) (*cloudiot.BindDeviceToGatewayResponse, error) {
+	client, err := getClient()
+
+	parent := fmt.Sprintf("projects/%s/locations/%s/registries/%s", projectID, region, registryID)
+	bindRequest := &cloudiot.BindDeviceToGatewayRequest{
+		DeviceId:  deviceID,
+		GatewayId: gatewayID,
+	}
+
+	response, err := client.Projects.Locations.Registries.BindDeviceToGateway(parent, bindRequest).Do()
+
+	if err != nil {
+		return response, err
```

Comment:
Might want to add additional context. The usual pattern is:
```suggestion
		return nil, fmt.Errorf("BindDeviceToGateway: %v", err)
```